blocks/store/selectors.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import { find } from 'lodash';
+
+/**
+ * Returns all the available block types.
+ *
+ * @param {Object} state Data state.
+ *
+ * @return {Array} Block Types.
+ */
+export function getBlockTypes( state ) {
+	return state.blockTypes.types;
+}
+
+export function getBlockType( state, name ) {
```

Comment:
JSDoc.

Reply:
We should optimize the reducer state to enable easier lookup by name (i.e. shape as object with block name as key).

Reply:
Not certain if it's a valid thing we want (keeping the order of the registered blocks) but that was the reasoning behind state as array.

Reply:
Actually, it looks like we we're not keeping the same order previously since we were using an object variable to hold the block list, I'm just going to use an object for now and let's see later if we really want to keep the order.